### PR TITLE
Reorganize files

### LINKED ISSUE #5: Refactor files into own concerns directories
We're developing a hierarchy that warrants splitting up files up and grouping similar things together. 

My first impulse was to group them into a `models/` and `views/`, but then it occurred to me that it also makes sense to group them into `resources/`, `items/`, `industry/`, etc.

I don't know if the latter is a good idea or not (either in an absolute sense, or as compared to the views/models duality), but this whole thing is an experiment, so I'm gonna run with it.

### PR DESCRIPTION
**Pull request by jholman**
Resolves #5

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,8 +5,15 @@
   "author": "garrettgsb",
   "license": "MIT",
   "dependencies": {
```
Comment (garrettgsb) [inline package.json]: What do these do?

File: package.json [hunk 2]
```diff
@@ -5,8 +5,15 @@
...
+    "babel-plugin-transform-decorators-legacy": "^1.3.5",
+    "babel-preset-env": "^1.7.0",
+    "babel-preset-stage-1": "^6.24.1",
+    "css-loader": "^0.28.11",
+    "node-sass": "^4.9.0",
     "react": "^16.3.2",
-    "react-dom": "^16.3.2"
+    "react-dom": "^16.3.2",
```
Comment (garrettgsb) [inline package.json]: Doesn't `react-dom` come with `react`? Why do we need it here?

File: src/public/index.html [hunk 3]
```diff
@@ -3,7 +3,6 @@
   <head>
     <meta charset="utf-8">
     <title>⾦</title>
-    <link rel="stylesheet" href="style.css">
```
Comment (garrettgsb) [inline src/public/index.html]: ...?

File: .babelrc [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (garrettgsb) [inline .babelrc]: Can you talk through what this does?

Comment (jholman) [reply .babelrc]: I cannot talk through it in any detail.  presets and plugins seem to configure babel's behaviour to know about a larger range of transformations that are legal.

* `env` is their suggested default set.
* `react` is react, duh
* `stage-1` is whatever TC39 proposals have currently reached at least stage-1 approval, out of 4 stages, so this is all the experimental shit
* `transform-class-properties` might actually be included in `stage-1`, but it's the thing that lets you do use arrow functions for methods, and has other uninteresting implications
* `transform-strict-mode`, I doubt this is actually necessary, hmn
* `transform-decorators-legacy` enables the stage-2 decorators proposal, which should be in `stage-1`, but they took it out for political reasons

Can you tell I copy-pasted some of these from somewhere else, and then added one that I wanted for my own reasons?  :D

File: package.json (hunk 1 above)
Comment (jholman) [reply package.json]: They all enable babel stuff, configured in `.babelrc`, or the webpack CSS stuff, enabled in the webpack config.

File: package.json (hunk 2 above)
Comment (jholman) [reply package.json]: It does not, AFAIK.  This is why `index.js` contains the line `import ReactDOM from 'react-dom';`

File: src/public/index.html (hunk 3 above)
Comment (jholman) [reply src/public/index.html]: Moved to `app.js`, to enable hot-reloading (which also required some dependencies and some fiddling with `webpack.config.js`

File: .babelrc (hunk 4 above)
Comment (garrettgsb) [reply .babelrc]: Okay, neat. LGTM.

**Pull request merged**